app/services/BusinessReportingMethodService.scala

```diff
@@ -0,0 +1,158 @@
+import models.core.Nino
+import models.incomeSourceDetails.viewmodels.BusinessReportingMethodViewModel
+import models.incomeSourceDetails.{IncomeSourceDetailsError, IncomeSourceDetailsModel, LatencyDetails}
+import models.itsaStatus.ITSAStatusResponseError
+import models.updateIncomeSource.{TaxYearSpecific, UpdateIncomeSourceResponse}
+import play.api.Logger
+import uk.gov.hmrc.http.{HeaderCarrier, InternalServerException}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class BusinessReportingMethodService @Inject()(incomeTaxViewChangeConnector: IncomeTaxViewChangeConnector,
+                                               dateService: DateService,
+                                               implicit val appConfig: FrontendAppConfig) extends FeatureSwitching {
+
+  def checkITSAStatusCurrentYear(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Boolean] = {
+    val yearEnd = dateService.getCurrentTaxYearEnd(isEnabled(TimeMachineAddYear)).toString.substring(2).toInt
+    val yearStart = yearEnd - 1
+
+    incomeTaxViewChangeConnector.getITSAStatusDetail(
+      nino = user.nino,
+      taxYear = s"$yearStart-$yearEnd",
+      futureYears = false,
+      history = false).flatMap {
+      case Right(itsaStatus) =>
+        val isMandatedOrVoluntary = itsaStatus.exists(_.itsaStatusDetails.exists(_.exists(_.isMandatedOrVoluntary)))
+        Future.successful(isMandatedOrVoluntary)
+      case Left(x: ITSAStatusResponseError) =>
+        Logger("application").error(s"[BusinessReportingMethodService][checkITSAStatusCurrentYear] $x")
+        Future.failed(new InternalServerException("[BusinessReportingMethodService][checkITSAStatusCurrentYear] - Failed to retrieve ITSAStatus"))
+    }
+  }
+
+  def isTaxYearCrystallised(ty: Int)(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Option[Boolean]] = {
+    if (ty <= 2023) {
+      incomeTaxViewChangeConnector.getLegacyCalculationList(Nino(user.nino), ty.toString).flatMap {
+        case res: CalculationListModel => Future.successful(res.crystallised)
+        case err: CalculationListErrorModel => Future.failed(new InternalServerException(err.message))
+      }
+    } else {
+      val tyRange = s"${(ty - 1).toString.substring(2)}-${ty.toString.substring(2)}"
+      incomeTaxViewChangeConnector.getCalculationList(Nino(user.nino), tyRange).flatMap {
+        case res: CalculationListModel => Future.successful(res.crystallised)
+        case err: CalculationListErrorModel => Future.failed(new InternalServerException(err.message))
+      }
+    }
+
+  }
+
+  def getBusinessReportingMethodDetails(incomeSourceId: String)(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Option[BusinessReportingMethodViewModel]] = {
+    incomeTaxViewChangeConnector.getBusinessDetails(user.nino).flatMap {
```

Comment:
we shouldn't need to get the income source details again - we already do it... twice!!